velox/vector/BaseVector.h

```diff
@@ -177,6 +177,14 @@ class BaseVector {
     return type_;
   }
 
+  void setType(TypePtr type) {
```

Comment:
Let's document this method. We need to clarify that the new type can be a different logical type, but must use the same physical type. Also, need to clarify that this is a mutating method and therefore the caller must ensure that the vector is writable, e.g. is singly-referenced.

Pass type as const & or use std::move in type_ = std::move(type).

Reply:
@mbasmanova Added to document for this method.